File: server/public/tools/elements/widgets/GtBox.ts
import {PolymerElement} from "../../polymer/PolymerElement";
import {Property, Element} from "../../polymer/Annotations";

@Element({
	selector: "gt-box",
	template: "/assets/imports/box.html"
})
export class GtBox extends PolymerElement {
	@Property
	public heading: string;
	@Property
	public scrollable: string;

	@Property({reflect: true})
	public dark: boolean;
}

@Element({
	selector: "gt-alert",
	template: "/assets/imports/box.html",
	dependencies: [GtBox]
})
export class GtAlert extends PolymerElement {
	@Property public icon: String;

	@Property({ reflect: true })
	public dark: boolean;
}

File: client/src/gt/components/widget/GtBox.scala
package gt.components.widget

import gt.components.GtHandler
import utils.jsannotation.js
import xuen.Component

object GtBox extends Component[GtBox](
	selector = "gt-box",
	templateUrl = "/assets/imports/box.html"
)

@js class GtBox extends GtHandler {
	val heading = attribute[String]
	val scrollable = attribute[Boolean]
	val dark = attribute[Boolean]

	val hasHeading = heading ~ { _ != null }
}
